src/parser/wsdl.js

```diff
@@ -104,8 +106,26 @@ class WSDL {
         self.xmlnsInEnvelope = self._xmlnsMap();
         callback(err, self);
       });
+    }
+
+    if (self.syncLoad) {
+      loadUpSchemas();
+    } else {
+      process.nextTick(loadUpSchemas);
+    }
+  }
 
+  loadSync() {
+    this.syncLoad = true;
+    var result;
+    this.load(function (err, wsdl) {
+        result = wsdl;
     });
+    // This is not intuitive but works as by setting syncLoad to true means this
+    // load function and its callback all are executed before the loadSync function
+    // returns. The outcome here is that the following result is always set correctly.
+    this.syncLoad = false;
```

Comment:
I don't think `this.syncLoad` is concurrency safe. Please note that the async `WSDL.load` will take a few eventloop ticks to finish and `WSDL.loadSync` can be invoked in between which causes an inconsistent `syncLoad` state.

IMO, `syncLoad` should be an argument. We can refactor the common logic to a `_loadAsyncOrSync` with `sync` as an argument.